mappings/net/minecraft/block/BlockState.mapping

```diff
@@ -1,54 +1,205 @@
 CLASS bom net/minecraft/block/BlockState
-	CLASS bom$a
+	CLASS bom$a ShapeCache
 		FIELD a DIRECTIONS [Ley;
+		FIELD b cull Z
+		FIELD f shapes [Lclt;
+		METHOD <init> (Lbom;)V
+			ARG 1 state
+			ARG 2 block
+			ARG 3 shape
+		METHOD c (Lbom$a;)[Lclt;
+	FIELD c shapeCache Lbom$a;
 	FIELD d luminance I
+	METHOD <init> (Lbft;Lcom/google/common/collect/ImmutableMap;)V
+		ARG 1 block
+		ARG 2 propertyMap
 	METHOD a allowsSpawning (Lagk;)Z
+		ARG 1 entity
 	METHOD a calcBlockBreakingDelta (Laql;Lbae;Let;)F
+		ARG 1 player
+		ARG 2 view
+		ARG 3 pos
+	METHOD a (Laty;)Z
+		ARG 1 ctx
 	METHOD a getBlockBrightness (Lbac;Let;)I
+		ARG 1 view
+		ARG 2 pos
+	METHOD a (Lbae;Let;)Z
+		ARG 1 view
+		ARG 2 pos
 	METHOD a canPlaceAtSide (Lbae;Let;Lcgh;)Z
+		ARG 1 view
+		ARG 2 pos
+		ARG 3 env
+	METHOD a (Lbae;Let;Lcle;)Lclt;
+		ARG 1 view
+		ARG 2 pos
+		ARG 3 ePos
+	METHOD a getCullShape (Lbae;Let;Ley;)Lclt;
```

Comment:
Too vague.

Reply:
Nonono. What's the shape of the VoxelShapeContainer? I'm curious if it's just an outline of a face or something.

Reply:
Oh, right.

So it calls `return VoxelShapes.a(this.m(view, pos), facing);`. For the default implementation, `this.m(view, pos)` calls getBoundingShape which returns VoxelShapes.fullCube() by default. From what it looks like, `VoxelShapes.a` takes that shape and offsets it in some way (returns an instance of `clr`, which extends `VoxelShapeContainer`), or returns an empty shape if the shape doesn't touch the edge of the block on that face

Reply:
Unrelated to implementation, but should this name maybe mention occlusion?